superset-frontend/src/components/Select/utils.ts

```diff
@@ -60,7 +61,19 @@ export function findValue<OptionType extends OptionTypeBase>(
   return (Array.isArray(value) ? value : [value]).map(find);
 }
 
-export function hasOption(search: string, options: AntdOptionsType) {
+export function hasOption<VT extends string | number>(
+  value: VT,
+  options?: VT | VT[] | { value: VT } | { value: VT }[],
+) {
+  const optionsArray = ensureIsArray(options);
+  return (
+    optionsArray.find(x =>
+      typeof x === 'object' ? x.value === value : x === value,
+    ) !== undefined
```

Comment:
```suggestion
    optionsArray.find(option =>
      typeof option === 'object' ? option.value === value : option === value,
    ) !== undefined
```

Reply:
In general I'd prefer to use `x` in those one liners because sometimes you may want to rename the iterable variable as well.